shared/constants/metametrics.js

```diff
@@ -164,6 +164,8 @@
  * identify the new number_of_accounts trait
  * @property {string} [THREE_BOX_ENABLED] - when 3box feature is toggled we
  * identify the 3box_enabled trait
+ * @property {string} [NUMBER_OF_NFT_COLLECTIONS] - user trait for number of
```

Comment:
```suggestion
 * @property {'number_of_nft_collections'} [NUMBER_OF_NFT_COLLECTIONS] - user trait for number of
```

New change. Replacing the type with the string literal as mentioned by Brad in this comment. 
https://github.com/MetaMask/metamask-extension/pull/14367#discussion_r845270295